firmware/arduino/chessboard/movementFunctions.ino

```diff
@@ -37,3 +37,37 @@ void centerPiece()
     // Turn electromagnet off
     digitalWrite(ELECTROMAGNET, LOW);
 }
+
+// The handler function that acts on different error codes output by movement primitives
+bool statusCodeHandler(uint8_t status)
+{
+    if (status == HIT_POS_X_ENDSTOP)
+    {
+        alignAxis(xMotor, MAX_POSITION);
+        return true;
+    }
+    else if (status == HIT_POS_Y_ENDSTOP)
+    {
+        alignAxis(yMotor, MAX_POSITION);
+        return true;
+    }
+    else if (status == HIT_NEG_X_ENDSTOP)
+    {
+        alignAxis(xMotor, ZERO_POSITION);
+        return true;
+    }
+    else if (status == HIT_NEG_Y_ENDSTOP)
+    {
+        alignAxis(yMotor, ZERO_POSITION);
+        return true;
+    }
+    else if (status == INVALID_ALIGNMENT)
+    {
+        home();
+        return true;
+    }
+    else if (status == INVALID_ARGS)
+    {
+        return false;
+    }
+}
```

Comment:
The reason you're getting that error and the slight optimization I have fortunately have the same fix. Notice that if you pass in a `status` that isn't checked for, the function can't actually return a bool, which is what it's expecting, that's where your error is coming from. The slight optimization is this. See how it's mostly returning true except for one case? You can make the function return true by default at the bottom, and if you ever meet the conditions to return false, you do that before you arrive at the return true statement.

Reply:
This is the structure I'm going for:
```
bool foo ()
{
  if (condition1)
  {
    action1
  }
  else if (condition2)
  {
    action1
  }
  else if (condition3)
  {
    action1
    return false;
  }

  return true;
}
```

Reply:
This way, you always return something and avoid the repeated `return true;` statements :)